File: Vector2D.cs
public struct Vector2D{
  public float X;
  public float Y;

  public Vector2D{
    this.X = x;
    this.Y = y;
  }
  // add operator overload.
  public static Vector2D operator +(Vector2D a, Vector2D b){
    return new Vector2D(a.X + b.X, a.Y + b.Y);
  }
  /*
  public static Vector2D plus(Vector2D a, Vector2D b){
    return new Vector2D(a.X + b.X, a.Y + b.Y);
  }
  */
}

File: Vector2D.cpp
struct Vector2D{
    float X;
    float Y;

    Vector2D(float x, float y){
        this->X = x;
        this->Y = y;
    }

    static const Vector2D Zero;

    // Operator overloads
    Vector2D operator+(const Vector2D &v) const {
        return Vector2D(X + v.X, Y + v.Y);
    }

    Vector2D operator-(const Vector2D &v) const {
        return Vector2D(X - v.X, Y - v.Y);
    }

    Vector2D operator-() const {
        return Vector2D(-X, -Y);
    }

    Vector2D operator*(float s) const {
        return Vector2D(X * s, Y * s);
    }

    Vector2D operator/(float s) const {
        return Vector2D(X / s, Y / s);
    }
};

const Vector2D Vector2D::Zero(0,0);